packages/@react-stately/combobox/src/useComboBoxState.ts

```diff
@@ -174,6 +165,15 @@ export function useComboBoxState<T extends object>(props: ComboBoxStateProps<T>)
       lastValue.current = inputValue;
     }
 
+    // Close when an item is selected, if open state or selectedKey is uncontrolled.
+    if (
+      selectedKey != null &&
+      selectedKey !== lastSelectedKey.current &&
+      (props.isOpen === undefined || props.selectedKey === undefined)
+    ) {
+      triggerState.close();
+    }
```

Comment:
Relocated this since it fixes a particular flow controlled inputValue + controlled open state:
Previously:
1. User focuses and hits Enter to select a option
2. This block fires since it was higher up, closing the menu
3. This [block](https://github.com/adobe/react-spectrum/pull/1587/files#diff-52d3ae5861fe9f80aaec78778d0a666b2b9d10dbf2aac7133bd6d3d70cade0faR155-R166) fires, reseting the inputValue and causing `onInputChange` to fire.
4. `onInputChange` then sets isOpen to true, resulting in the menu being open when it should've closed